packages/near-api-js/src/utils/view-function.ts

```diff
@@ -0,0 +1,67 @@
+import { BlockReference, CodeResult } from '../providers/provider';
+import { printTxOutcomeLogs } from './logging';
+import { FunctionCallOptions } from '../account';
+import { Connection } from '../connection';
+
+function parseJsonFromRawResponse(response: Uint8Array): any {
+    return JSON.parse(Buffer.from(response).toString());
+}
+
+function bytesJsonStringify(input: any): Buffer {
+    return Buffer.from(JSON.stringify(input));
+}
+
+export interface ViewFunctionCallOptions extends FunctionCallOptions {
+    connection: Connection;
+    parse?: (response: Uint8Array) => any;
+    blockQuery?: BlockReference;
+}
+
+// TBD fix params
+/**
+ * Invoke a contract view function using the RPC API.
+ * @see [https://docs.near.org/api/rpc/contracts#call-a-contract-function](https://docs.near.org/api/rpc/contracts#call-a-contract-function)
+ *
+ * @param viewFunctionCallOptions.contractId NEAR account where the contract is deployed
+ * @param viewFunctionCallOptions.methodName The view-only method (no state mutations) name on the contract as it is written in the contract code
+ * @param viewFunctionCallOptions.args Any arguments to the view contract method, wrapped in JSON
+ * @param viewFunctionCallOptions.parse Parse the result of the call. Receives a Buffer (bytes array) and converts it to any object. By default result will be treated as json.
+ * @param viewFunctionCallOptions.stringify Convert input arguments into a bytes array. By default the input is treated as a JSON.
+ * @param viewFunctionCallOptions.jsContract Is contract from JS SDK, automatically encodes args from JS SDK to binary.
+ * @param viewFunctionCallOptions.blockQuery specifies which block to query state at. By default returns last "optimistic" block (i.e. not necessarily finalized).
+ * @returns {Promise<any>}
+ */
+export async function viewFunction({
```

Comment:
I agree that this needs to be refactored out of `Account` but that's not something that needs to be done as part of no longer requiring `Contract` to be instantiated with an `Account`. My main concern is that this carries too much baggage with it from `Account` and needs to be deconstructed rather than just moved in place.

As with `_changeMethod`, we can remove the need for a provided `Account` object and instantiate one when making the view call:
```js
return new Account(this.connection, 'dontcare').viewFunction({
```

Reply:
Totally agree, it should be part of different PR